sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -764,35 +894,119 @@ public Write withoutValidation() {
     }
 
     /**
-     * Returns a new {@link BigtableIO.Write} that will write using the given Cloud Bigtable service
-     * implementation.
+     * Returns a new {@link BigtableIO.Write} that will use an official Bigtable emulator.
      *
      * <p>This is used for testing.
+     */
+    @VisibleForTesting
+    public Write withEmulator(String emulatorHost) {
+      BigtableConfig config = getBigtableConfig();
+      return toBuilder().setBigtableConfig(config.withEmulator(emulatorHost)).build();
+    }
+
+    /**
+     * Returns a new {@link BigtableIO.Write} with the attempt timeout in milliseconds for writes.
      *
      * <p>Does not modify this object.
      */
-    Write withBigtableService(BigtableService bigtableService) {
-      BigtableConfig config = getBigtableConfig();
-      return toBuilder().setBigtableConfig(config.withBigtableService(bigtableService)).build();
+    public Write withAttemptTimeout(long timeoutMs) {
+      checkArgument(timeoutMs > 0, "attempt timeout must be positive");
+      BigtableWriteOptions options = getBigtableWriteOptions();
+      return toBuilder()
+          .setBigtableWriteOptions(options.toBuilder().setAttemptTimeout(timeoutMs).build())
+          .build();
     }
 
     /**
-     * Returns a new {@link BigtableIO.Write} that will use an official Bigtable emulator.
+     * Returns a new {@link BigtableIO.Write} with the operation timeout in milliseconds for writes.
      *
-     * <p>This is used for testing.
+     * <p>Does not modify this object.
+     */
+    public Write withOperationTimeout(long timeoutMs) {
+      checkArgument(timeoutMs > 0, "operation timeout must be positive");
+      BigtableWriteOptions options = getBigtableWriteOptions();
+      return toBuilder()
+          .setBigtableWriteOptions(options.toBuilder().setOperationTimeout(timeoutMs).build())
+          .build();
+    }
+
+    /**
+     * Returns a new {@link BigtableIO.Write} with the retry delay in milliseconds.
```

Comment:
This is the public surface that end users will interact with, please provide more details what retry delay is and when its used or why anyone cares :)

Reply:
similar for all of the other new options